### PR TITLE
Fix wca_excerpt case sensitivity

### LINKED ISSUE #854: wca_excerpt doesn't handle case insensitivity
@jonatanklosko added some awesome improvements to our search and our `wca_excerpt` method in #804, but unfortunately it looks like it doesn't handle case insensitivity. For example:

https://www.worldcubeassociation.org/search?q=Mateusz%20Wojciech
image

https://www.worldcubeassociation.org/search?q=mateusz%20Wojciech
image

This would make a good ment1 project because you get to write a test!

### PR DESCRIPTION
**Pull request by jonatanklosko**
Fixes #854.
@jfly good catch!

### CODE REVIEW

File: WcaOnRails/spec/helpers/application_helper_spec.rb [hunk 1]
```diff
@@ -34,5 +34,11 @@
       expected = "...t with a super word in the middle (directly here) <strong>followed</strong> by the rest of an awesome and peculiar sentence. What do you <strong>think</strong>?"
       expect(helper.wca_excerpt(text, ["followed", "think"])).to eq expected
     end
+
+    it "is case insensitive" do
+      text = "Some #{("thing " * 15)}match #{("thing " * 15)}end."
+      expected = "<strong>Some</strong> #{("thing " * 15)}<strong>match</strong> #{("thing " * 15)}<strong>end</strong>."
+      expect(helper.wca_excerpt(text, ["some", "match", "END"])).to eq expected
+    end
```
Comment (jonatanklosko) [inline WcaOnRails/spec/helpers/application_helper_spec.rb]: This looks kinda weird so, see the output before the fix:
```
1) ApplicationHelper#wca_excerpt is case insensitive
   Failure/Error: expect(helper.wca_excerpt(text, ["some", "match", "END"])).to eq expected
     
     expected: "<strong>Some</strong> thing thing thing thing thing thing thing thing thing thing thing thing thing thing thing <strong>match</strong> thing thing thing thing thing thing thing thing thing thing thing thing thing thing thing <strong>end</strong>."
          got: "...g thing thing thing thing thing thing thing thing <strong>match</strong> thing thing thing thing thing thing thing thing th..."
```

File: WcaOnRails/spec/helpers/application_helper_spec.rb [hunk 2]
```diff
@@ -34,5 +34,12 @@
       expected = "...t with a super word in the middle (directly here) <strong>followed</strong> by the rest of an awesome and peculiar sentence. What do you <strong>think</strong>?"
       expect(helper.wca_excerpt(text, ["followed", "think"])).to eq expected
     end
+
+    it "is case insensitive" do
+      devider = (["thing"] * 15).join(' ')
+      text = "Some #{devider} match #{devider} end."
+      expected = "<strong>Some</strong> #{devider} <strong>match</strong> #{devider} <strong>end</strong>."
+      expect(helper.wca_excerpt(text, ["some", "match", "END"])).to eq expected
+    end
```
Comment (jonatanklosko) [inline WcaOnRails/spec/helpers/application_helper_spec.rb]: This looks kinda weird so see the output before the fix
```
1) ApplicationHelper#wca_excerpt is case insensitive
     Failure/Error: expect(helper.wca_excerpt(text, ["some", "match", "END"])).to eq expected
       
       expected: "<strong>Some</strong> thing thing thing thing thing thing thing thing thing thing thing thing thing thing thing <strong>match</strong> thing thing thing thing thing thing thing thing thing thing thing thing thing thing thing <strong>end</strong>."
            got: "...g thing thing thing thing thing thing thing thing <strong>match</strong> thing thing thing thing thing thing thing thing th..."
```

File: WcaOnRails/spec/helpers/application_helper_spec.rb [hunk 3]
```diff
@@ -32,7 +32,14 @@
     it "handles multiple phrases correctly highlighting them" do
       text = "A long text with a super word in the middle (directly here) followed by the rest of an awesome and peculiar sentence. What do you think?"
       expected = "...t with a super word in the middle (directly here) <strong>followed</strong> by the rest of an awesome and peculiar sentence. What do you <strong>think</strong>?"
-      expect(helper.wca_excerpt(text, ["followed", "think"])).to eq expected
+      expect(helper.wca_excerpt(text, %w(followed think))).to eq expected
+    end
+
+    it "is case insensitive" do
+      devider = (["thing"] * 15).join(' ')
```
Comment (jfly) [inline WcaOnRails/spec/helpers/application_helper_spec.rb]: nit: `devider` -> `divider`

File: WcaOnRails/spec/helpers/application_helper_spec.rb [hunk 4]
```diff
@@ -32,7 +32,14 @@
     it "handles multiple phrases correctly highlighting them" do
       text = "A long text with a super word in the middle (directly here) followed by the rest of an awesome and peculiar sentence. What do you think?"
       expected = "...t with a super word in the middle (directly here) <strong>followed</strong> by the rest of an awesome and peculiar sentence. What do you <strong>think</strong>?"
-      expect(helper.wca_excerpt(text, ["followed", "think"])).to eq expected
+      expect(helper.wca_excerpt(text, %w(followed think))).to eq expected
+    end
+
+    it "is case insensitive" do
```
Comment (jfly) [inline WcaOnRails/spec/helpers/application_helper_spec.rb]: Philosophy thing: I probably would have folded this test into the previous test. I suppose it's a tradeoff between lots of small, simple tests, and paranoia about test runtime. I'm okay with whatever you prefer here.

**Comment by jfly** [comment]
LGTM!

File: WcaOnRails/spec/helpers/application_helper_spec.rb (hunk 4 above)
Comment (jonatanklosko) [inline WcaOnRails/spec/helpers/application_helper_spec.rb]: I hear you, anyway I really prefer the 'small, simple tests' approach. If we'd think about runtime and about less code (sometimes we can easily test multiple things at once), we could end up with long tests, and the idea behind them is hard to figure out. IMO ideally each test has good description and the code does exactly what the description says. For example when `it "works"` fails we usually need to look into the code anyway.
Especially, this test is not that obvious even now as it test kinda tricky thing.

**Pull request merged**